### ISSUE TITLE
Limit peer review to users above level 10

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#2852]
Peer review can not be done by a beginner. It takes quite some experience to know what is on-topic and off-topic, what is a copyright violation or not.

So, I suggest hiding the navigation drawer's "Review" item from users who have not yet reached level 10.

I chose 10 because it is a round number, but I am open to other suggestions.

This means storing the level somewhere when it is retrieved from the server (in the Achievements activity). People who have never opened the Achievements activity will not see the "Review" item, because their level is unknown.

**Comment by madhurgupta10** [comment]
@nicolas-raoul Instead of hiding, what if we have a dialog which tells the user why he/she can't review, this will motivate him/her to reach next levels quickly :)

**Comment by neslihanturan** [comment]
>This means storing the level somewhere when it is retrieved from the server (in the Achievements activity). People who have never opened the Achievements activity will not see the "Review" item, because their level is unknown.

If we will implement this, we can calculate achievements on background on application start. Or as @madhurgupta10 said, we can inform user. We can first "if you like to discover review item please check your achievement score first, and redirect user to achievements", on next visit if they are under the score inform that why we have such mechanism. If they exceed, let them review. 

But this should be designed carefully because saying "you are not eligible to review" may be understood as a rude/annoying expression.

**Comment by nicolas-raoul** [comment]
@madhurgupta10 Indeed, displaying the button and showing an explanation can be a way to motivate people, indeed. We don't want to make people play the system, though, for instance by uploading 1000 picture of a mushroom and embedding them all in sub-pages of the Swahili Wikipedia, for instance.

@neslihanturan  You are right. I am confident that we can do this in a careful way. StackOverflow has tons of level-specific privileges and manages to not make them feel rude. https://stackoverflow.com/help/privileges

**Comment by misaochan** [comment]
Good idea! But umm, is level 10 a bit high? Maybe 5 or something...? I think you are the only one here who has level 10+, @nicolas-raoul . ;)

.

**Comment by nicolas-raoul** [comment]
Yes sure, 5 is probably enough.
But if it is only to allow developers to performs tests, then maybe implementing https://github.com/commons-app/apps-android-commons/issues/2861 is safer strategy? :-)

**Comment by misaochan** [comment]
Not really for the developers, I just thought that 10 is a bit too high. :) Devs can always hardcode or mock their level with the debugger.

**Comment by madhurgupta10** [comment]
@nicolas-raoul @misaochan Can I start implementing this?

**Comment by nicolas-raoul** [comment]
@madhurgupta10 Sure, thanks! :-)

**Comment by madhurgupta10** [comment]
@nicolas-raoul 
I just need a few clarifications!

When the user visits the Review Activity then we will check if the user level is >=5 if yes then we will let them review. If not then we will display a popup that explains that the user can't review until the profile reaches level 5 and redirects them to Achievements Activity. 

When the user visits the Review Activity again and he/she is still under level then we will display the same popup or something else or we will let them review?

**Comment by nicolas-raoul** [comment]
Good question!
The best would be to at first never display the Review item in the left menu, a bit like we don't display it for users who are not logged in.
Then each time the user goes to their Achievements and gets their level calculated, you would "unlocked" the feature (allow the review item to appear in the menu, either immediately or next time they open the app).

**Comment by madhurgupta10** [comment]
@nicolas-raoul Thanks, on it!

**Comment by nicolas-raoul** [comment]
Variation on this idea:
- Let all users do review
- When a review is done, send it:
  - If the user is over Level 5: to the wiki pages as currently done
  - If the user is below Level 5: to a "flagged" queue on our toolserver
- When a user over Level 5 wants to perform a review, first pull from that "flagged" queue, and only pull from normal if the "flagged" queue is empty.

Problem: It increases the time it takes to get bad images down. So here is a bonus idea:
- For images in the flagged queue, make 2 more users below Level 5 review it, and if more than 24 hours have passed without a user over Level 5 reviewing it.

If toolserver is down or not working, just post the review directly (like currently).

When a user below Level 5 uses "Nominate for deletion" (in Explore>Media details), the picture could be sent to that "flagged" queue.

**Comment by nicolas-raoul** [comment]
Switching to high priority as Commons admins are getting frustrated with erroneous deletion requests from users new to Commons.

**Comment by Jason-Whitmore** [comment]
I just came across this issue from the recent mention of it and I am interested. However, I have a question about a possible implementation:

Assuming that some sort of system is designed to combine reviews from low and high level users, wouldn't that system be implemented on the server side and not on the app (client)?

Also, in a worst case scenario (admins are *really* frustrated), we could do what was originally thought and just remove the review functionality entirely for low level users.
